firmware/controllers/tcu.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ * @file        tcu.cpp
+ * @brief       Base classes for gear selection and transmission control
+ *
+ * @date Aug 31, 2020
+ * @author David Holdeman, (c) 2020
+ */
+
+#include "tcu.h"
+#include "tunerstudio_outputs.h"
+
+void TransmissionControllerBase::update(gear_e gear) {
+    setCurrentGear(gear);
+    postState();
+}
+
+gear_e TransmissionControllerBase::setCurrentGear(gear_e gear) {
+    currentGear = gear;
+    return getCurrentGear();
+}
+
+gear_e TransmissionControllerBase::getCurrentGear() {
+    return currentGear;
+}
+
+void TransmissionControllerBase::postState() {
+#if EFI_TUNER_STUDIO
+    tsOutputChannels.tcuCurrentGear = getCurrentGear();
+#endif
+}
+
+gear_e GearControllerBase::getDesiredGear() {
```

Comment:
in c++, functions that don't permute internal state can be declared `const`

see more here: https://www.tutorialspoint.com/const-member-functions-in-cplusplus